src/components/TrialRatingScale.tsx

```diff
@@ -17,20 +18,44 @@ export const TrialRatingScale: React.FunctionComponent<TrialRatingScale> = ({
 }) => {
   // Keep track of selected value
   const ratingOptions = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+  const [locked, setLocked] = useState(false)
   const [currentButton, setCurrentButton] = useState<number>()
 
   // Calculate size of buttons
   const screenWidth = Dimensions.get('screen').width
   const boxSize = 0.1 * screenWidth
 
+  // Update internal state + parent
+  const setSelectionOption = (value: number, debounceLock = false) => {
+    setCurrentButton(value)
+    // If we are swiping we only want to lock when we stop swiping
+    if (debounceLock) {
+      debouncedLock()
+    } else {
+      setLocked(true)
+      onChange(value)
+    }
+  }
+
+  // Debounce locking for 100ms
+  const debouncedLock = useCallback(
```

Comment:
Doesn't this need to be defined before line 33? Otherwise there could be a `ReferenceError`

Reply:
There shouldn't be due to how hoisting works in JS